sqldelight-studio-plugin/src/main/kotlin/com/squareup/sqldelight/intellij/lang/SqlDelightFileViewProviderFactory.kt

```diff
@@ -16,95 +16,85 @@
+    if (virtualFile.moduleDirectory() == null) {
+      // .sq file is not under src/variant/sqldelight.
+      return SingleRootFileViewProvider(psiManager, virtualFile, eventSystemEnabled)
+    }
     return SqlDelightFileViewProvider(virtualFile, language, psiManager, eventSystemEnabled)
   }
 }
 
-internal class SqlDelightFileViewProvider(virtualFile: VirtualFile, language: Language,
-    val psiManager: PsiManager, eventSystemEnabled: Boolean) :
-    SingleRootFileViewProvider(psiManager, virtualFile, eventSystemEnabled, language) {
+internal class SqlDelightFileViewProvider(
+    virtualFile: VirtualFile,
+    language: Language,
+    psiManager: PsiManager,
+    eventSystemEnabled: Boolean
+) : SingleRootFileViewProvider(psiManager, virtualFile, eventSystemEnabled, language) {
 
-  val documentManager = PsiDocumentManager.getInstance(psiManager.project)
   val file: SqliteFile by lazy {
     getPsiInner(SqliteLanguage.INSTANCE) as SqliteFile
   }
 
   override fun contentsSynchronized() {
     super.contentsSynchronized()
-    documentManager.performWhenAllCommitted { generateJavaInterface(true) }
+    generateJavaInterface(true)
   }
 
   internal fun generateJavaInterface(fromEdit: Boolean = false) {
-    val manager = SqlDelightManager.getInstance(file) ?: return
-
     // Mark the file as dirty and re-parse.
     file.dirty = true
     file.parseThen({ parsed ->
-      manager.symbolTable += SymbolTable(parsed, virtualFile, parsed.relativePath())
-      manager.setDependencies(this) {
-        file.status = sqldelightValidator.validate(parsed.relativePath(),
-            parsed, manager.symbolTable)
-        (file.status as Status.ValidationStatus).dependencies.filter { it != virtualFile }
-      }
-      manager.triggerDependencyRefresh(virtualFile, fromEdit)
-
-      if (file.status is Invalid) return@parseThen
-
-      file.status = SqliteCompiler.write(parsed,
-          (file.status as Status.ValidationStatus.Validated).queries,
-          parsed.relativePath(),
-          virtualFile.getPlatformSpecificPath().moduleDirectory(parsed) + File.separatorChar
-      )
-
-      if (file.status is Status.Success) {
-        val generatedFile = localFileSystem.findFileByIoFile((file.status as Status.Success).generatedFile)
-        if (generatedFile == null) {
-          Logger.getInstance(SqlDelightFileViewProvider::class.java)
-              .debug("Failed to find the generated file for ${file.virtualFile.path}, " +
-                  "it currently is ${file.generatedFile?.virtualFile?.path}")
-          return@parseThen
+      var status: Status = validate(parsed, fromEdit) ?: return@parseThen
+      if (status is ValidationStatus.Validated) {
+        status = parsed.compile(status)
+        if (status is Status.Success) {
+          val typeString = StringBuilder()
+          JavaFile.builder(file.relativePath.pathPackage(), status.model).build().writeTo(typeString)
```

Comment:
You can just call toString on a JavaFile if you want a string. Otherwise you should he able to stream to the file.